test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignedPackageIntegrityVerificationTests.cs

```diff
@@ -952,33 +952,30 @@ public async Task VerifyPackageContentHash_SignedPackagesAsync()
             // Arrange
             var nupkg = new SimpleTestPackageContext();
 
-            string expectedHash = null;
-
-            using (var zipStream = await nupkg.CreateAsStreamAsync())
+            using (MemoryStream packageStream = await nupkg.CreateAsStreamAsync())
             {
-                expectedHash = Convert.ToBase64String(new CryptoHashProvider("SHA512").CalculateHash(zipStream));
-            }
+                string expectedHash = Convert.ToBase64String(new CryptoHashProvider("SHA512").CalculateHash(packageStream));
```

Comment:
I've seen cryptographic exceptions in past projects when some crypto related classes were garbage collected without being disposed, I just can't remember which classes had this behaviour. Perhaps we could prevent other flakiness if we make sure the `new CryptoHashProvider` gets disposed as well.

Reply:
[`CryptoHashProvider`](https://github.com/NuGet/NuGet.Client/blob/3803820961f4d61c06d07b179dab1d0439ec0d91/src/NuGet.Core/NuGet.Common/CryptoHashProvider.cs#L14) does not implement `IDisposable`, but internally `CryptoHashProvider.CalculateHash(Stream)` already [disposes](https://github.com/NuGet/NuGet.Client/blob/3803820961f4d61c06d07b179dab1d0439ec0d91/src/NuGet.Core/NuGet.Common/CryptoHashProvider.cs#L61) the `HashAlgorithm` instance it creates.